src/application/wizard/resolvers.ts

```diff
@@ -67,9 +67,24 @@ const resolvers = (wizard: WizardService, userService: UserService): IResolvers
             _,
             { id: submissionId, coverLetter }: { id: SubmissionId; coverLetter: string },
             context,
-        ): Promise<Submission | null> {
+        ): Promise<Submission> {
+            const user = await userService.getCurrentUser(context.authorizationHeader);
+            return await wizard.saveFilesPage(user, submissionId, coverLetter);
+        },
+
+        async saveDetailsPage(
+            _,
+            {
+                id: submissionId,
+                details,
+            }: {
+                id: SubmissionId;
+                details: ManuscriptDetails;
+            },
+            context,
+        ): Promise<Submission> {
             const user = await userService.getCurrentUser(context.authorizationHeader);
-            return wizard.saveFilesPage(user, submissionId, coverLetter);
+            return await wizard.saveDetailsPage(user, submissionId, details);
```

Comment:
as this returns a promise isnt the `await` here redundant?

Reply:
I wondered - @norisalsaadie seemed to think this was important as it was a resolver :man_shrugging:

Reply:
Im pretty sure that it wont do anything bad it just is implied by the function signature returning a promise and being async

Reply:
Done some reading - and looking here all the resolvers do seem to wait for the data to come back before restructuring/returning... https://www.apollographql.com/docs/tutorial/resolvers/

Reply:
I think that will need a separate line of code, I don't think this will wait till the saveDetails is done, it will just return the promise that saveDetailsPage returns.
https://jakearchibald.com/2017/await-vs-return-vs-return-await/
https://eslint.org/docs/rules/no-return-await

Reply:
It might be redundant but I always prefer explicit awaits as it plugs the need for more in-depth knowledge of JS. But that's fine. I'll remove it.